int/tests/conftest.py

```diff
@@ -0,0 +1,166 @@
+"""Automate setup of Proxy Mediator + Mediated Agent:
+
+1. Retrive invitation from mediator
+2. Receive invitation in proxy
+3. Retrieve invitation from proxy
+4. Deliver to agent through Admin API
+5. Request mediation from proxy
+6. Set proxy as default mediator
+"""
+
+import asyncio
+from base64 import urlsafe_b64decode
+import json
+from os import getenv
+from functools import partial
+
+from acapy_client import Client
+from acapy_client.api.connection import (
+    get_connection,
+    receive_invitation,
+    create_invitation,
+)
+from acapy_client.api.mediation import (
+    get_mediation_requests_mediation_id,
+    post_mediation_request_conn_id,
+    put_mediation_mediation_id_default_mediator,
+    get_mediation_requests,
+)
+from acapy_client.models.conn_record import ConnRecord
+from acapy_client.models.create_invitation_request import CreateInvitationRequest
+from acapy_client.models.mediation_create_request import MediationCreateRequest
+from acapy_client.models.receive_invitation_request import ReceiveInvitationRequest
+from acapy_client.models.get_mediation_requests_state import GetMediationRequestsState
+from acapy_client.types import Unset
+
+from httpx import AsyncClient
+import pytest
+
+from . import record_state
+from . import poll_until_condition
+
+
+PROXY = getenv("PROXY", "http://proxy:3000")
+AGENT_BOB = getenv("AGENT_BOB", "http://agent_bob:4012")
+EXTERNAL_MEDIATOR = getenv("EXTERNAL_MEDIATOR", "http://external_mediator:4013")
+
+
+async def get_proxy_invite() -> dict:
+    async with AsyncClient() as client:
+        url = None
+        while url is None:
+            r = await client.get(f"{PROXY}/retrieve_agent_invitation")
+            url = r.json()["invitation_url"]
+            if not url:
+                await asyncio.sleep(1)
+        return json.loads(urlsafe_b64decode(url.split("c_i=")[1]))
```

Comment:
For the stability of the integration tests, we should use the `poll_until_condition` for this -- otherwise, we may end up in an infinite loop. This might look something like:

```suggestion
async def get_proxy_invite() -> dict:
    async with AsyncClient() as client:
        async def _retrieve(client):
            r = await client.get(f"{PROXY}/retrieve_agent_invitation")
            return r.json()["invitation_url"]
        url = await poll_until_condition(lambda url: url is not None, partial(_retrieve, client)
    return json.loads(urlsafe_b64decode(url.split("c_i=")[1]))
```